src/routes/dashboard/containers/__tests__/DashboardContainer.spec.js

```diff
@@ -0,0 +1,254 @@
+      depth: 2,
+      language: language.code,
+      city: city.code
+    })
+    expectLoadingIndicator(state)
+  })
+
+  it('should display loading indicator if switching languages lasts long enough', () => {
+    const state: StateType = prepareState(successfulRouteState, { switchingLanguage: true })
+    expectLoadingIndicator(state)
+  })
+
+  it('should display loading indicator if cities are loading long enough', () => {
+    const state: StateType = prepareState(successfulRouteState, { cities: { status: 'loading' } })
+    expectLoadingIndicator(state)
+  })
+
+  it('should display loading indicator if languages are loading long enough', () => {
+    const state: StateType = prepareState(successfulRouteState, { languages: null })
+    expectLoadingIndicator(state)
+  })
+
+  it('should display LanguageNotAvailable if the route has the corresponding status', () => {
+    const state = prepareState({
+      status: 'languageNotAvailable',
+      depth: 2,
+      city: city.code,
+      language: language.code,
+      allAvailableLanguages: new Map(languages.map(lng => [lng.code, `/${city.code}/${lng.code}`]))
+    })
+    const store = mockStore(state)
+    const navigation = createNavigationScreenPropMock()
+    navigation.state.key = 'route-id-0'
+    jest.doMock('../../../../modules/categories/components/Categories', () => MockDashboard)
+    const DashboardContainer = require('../DashboardContainer').default
+
+    const { getByText } = render(
+      <Provider store={store}><DashboardContainer navigation={navigation} /></Provider>
+    )
+    expect(getByText('chooseALanguage')).toBeTruthy()
+  })
+
+  it('should display Dashboard component if the state is ready', () => {
+    const state: StateType = prepareState(successfulRouteState)
+    const store = mockStore(state)
+    const navigation = createNavigationScreenPropMock()
+    navigation.state.key = 'route-id-0'
+    jest.doMock('../../components/Dashboard', () => MockDashboard)
+    const DashboardContainer = require('../DashboardContainer').default
+    const result = TestRenderer.create(
+      <Provider store={store}><DashboardContainer navigation={navigation} /></Provider>
+    )
+    const dashboardInstance = result.root.findByType(MockDashboard)
+    expect(dashboardInstance.props).toEqual({
+      cities: [city],
+      cityCode: city.code,
+      language: language.code,
+      navigateToCategory: expect.any(Function),
+      navigateToIntegreatUrl: expect.any(Function),
+      navigateToEvent: expect.any(Function),
+      navigateToExtras: expect.any(Function),
+      navigateToDashboard: expect.any(Function),
+      navigation,
+      resourceCache,
+      stateView: expect.any(CategoriesRouteStateView),
+      t: expect.any(Function),
+      theme: brightTheme
+    })
+    const stateView = dashboardInstance.props.stateView
+    expect([stateView.root(), stateView.children()]).toEqual([rootCategory, categoriesMap.getChildren(rootCategory)])
+  })
+})
```

Comment:
Much of duplicated code between `CategoriesContainer` and `DashboardContainer` as well as between `CategoriesContainer.spec` and `DashboardContainer.spec`, maybe come code parts could be  reused. Probably  not worth the effort though.

Reply:
Yeah, but that's because right now, the Dashboard's pretty much the same as the Categories route. Since this will change in the future, it makes no sense to reduce the duplicity here imo...